features/dd-sdk-android-trace/src/main/java/com/datadog/trace/core/CoreTracer.java

```diff
@@ -264,7 +265,7 @@ public static class CoreTracerBuilder {
 
         private Config config;
         private String serviceName;
-        private Writer writer;
+        private Writer writer = new NoOpWriter();
```

Comment:
because otherwise you cannot build a tracer without passing a writer and in case you do it will throw a NoOp. Before the builder had a default value but I removed that when cleaning up the code and decided to have it as nullable. During these unit tests refactor I realized that is better to have it as NoOp in case it is not provided to remove the noise in the tests.